drivers/usb/udc/udc_bflb_bl61x.c

```diff
@@ -0,0 +1,1480 @@
+/*
+ * Copyright (c) 2024-2025 MASSDRIVER EI (massdriver.space)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "udc_common.h"
+
+#include <string.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/usb/udc.h>
+#include <zephyr/cache.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(udc_bflb_bl61x, CONFIG_UDC_DRIVER_LOG_LEVEL);
+
+#include <soc.h>
+#include <bflb_soc.h>
+#include <glb_reg.h>
+#include <pds_reg.h>
+#include <bouffalolab/common/usb_v2_reg.h>
+
+#define USB_BL61X_SPEED_LOW	1
+#define USB_BL61X_SPEED_FULL	0
+#define USB_BL61X_SPEED_HIGH	2
+
+#define USB_BL61X_CAP_ISO	1
+#define USB_BL61X_CAP_BULK	2
+#define USB_BL61X_CAP_INT	3
+
+#define USB_BL61X_FX_X_MASK	0x3F
+#define USB_BL61X_FX_X_OFFSET	8
+
+#define USB_BL61X_XPS_X_OFFSET	4
+
+#define USB_BL61X_HSFIFOCAP	512
+
+#define USB_BL61X_EP_DIR_IN	0
+#define USB_BL61X_EP_DIR_OUT	1
+#define USB_BL61X_FIFO_DIR_OUT	0
+#define USB_BL61X_FIFO_DIR_IN	1
+#define USB_BL61X_FIFO_DIR_BID	2
+#define USB_BL61X_FIFO_EP_NONE	15
+
+#define USB_BL61X_TIMER_AFTER_RESET_HS (0x44C)
+#define USB_BL61X_TIMER_AFTER_RESET_FS (0x2710)
+
+#define USB_MCX_COMEND_INT (1 << 3)
+
+#define DT_DRV_COMPAT bflb_bl61x_usb
+
+struct udc_bflb_bl61x_config {
+	uint32_t base;
+	size_t num_of_eps;
+	void (*irq_enable_func)(const struct device *dev);
+	void (*irq_disable_func)(const struct device *dev);
+	struct udc_ep_config *ep_cfg_in;
+	struct udc_ep_config *ep_cfg_out;
+	int speed_idx;
+};
+
+struct udc_bflb_bl61x_data {
+	struct k_work work;
+	struct k_fifo fifo;
+	bool ep_in[5];
+	bool setup_received;
+};
+
+enum udc_bflb_bl61x_ev_type {
+	/* Trigger next transfer */
+	UBFBL61X_EVT_XFER,
+	/* packet dma complete  for ctrl fifo*/
+	UBFBL61X_EVT_CTRL_END,
+	/* packet dma complete for specific endpoint */
+	UBFBL61X_EVT_END,
+};
+
+struct udc_bflb_bl61x_ev {
+	void *reserved;
+	const struct device *dev;
+	uint8_t ep_addr;
+	enum udc_bflb_bl61x_ev_type event;
+};
+
+K_MEM_SLAB_DEFINE(udc_bflb_bl61x_ev_slab, sizeof(struct udc_bflb_bl61x_ev),
+		  CONFIG_UDC_BFLB_BL61X_EVENT_COUNT, sizeof(void *));
+
+static void udc_bflb_bl61x_ev_submit(const struct device *dev,
+				 const uint8_t ep_addr,
+				 const enum udc_bflb_bl61x_ev_type event)
+{
+	struct udc_bflb_bl61x_data *priv = udc_get_private(dev);
+	struct udc_bflb_bl61x_ev *ev;
+	int ret;
+
+	ret = k_mem_slab_alloc(&udc_bflb_bl61x_ev_slab, (void **)&ev, K_NO_WAIT);
+	if (ret) {
```

Comment:
I think the USB area started to anticipate a possible MISRA compliance, which wants `if (expr)` have `expr` be a bool (seen [here](https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html), row 85).

And switching from `ret` to `err` would be fitting the rest of USB codebase. Maybe this would be pointed at for next reviewers.

```suggestion
	int err;

	err = k_mem_slab_alloc(&udc_bflb_bl61x_ev_slab, (void **)&ev, K_NO_WAIT);
	if (err != 0) {
```

Reply:
done